### ISSUE TITLE
🚀 | Add resume option to `play_card` in future3

### ISSUE THREAD

**Issue by votti** [MiczFlor/RPi-Jukebox-RFID#1945]
### What functionality would you like to see in your phoniebox?

Also in future3 the resume functionality should be provided.
Eg some folders (audiobook) should be set to always resume from the previous position.

### How do you envision the feature to work from a users perspective?

Set in the configuration of a card to resume it and the next time switching from another card to this card, the playback should resume from the old position.

**Comment by plimptm** [comment]
With a month or two of real-life-with-kids experience after building, I've been thinking about this issue and related functionality 'tweaks' that stem from the dual use cases of 1) jukebox (music) and 2) audiobook player. Resume functionality is pretty critical for a manageable audiobook solution, and I've addressed that by installing [votti:future3/feature/resume](https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1946). 

As @votti mentioned when opening this issue, it would be great if audiobook folders could be set to automatically enable this function. It would also be great if other existing functions could have their behavior tweaked by the same logic (e.g. perhaps mapping the 'next track' and 'previous track' buttons to seek forwards/backwards a preset (but configurable) number of seconds.

Novice question - does the way the application current reads metadata about the contents of each folder help it to 'know' the difference between music and an audiobook? How difficult/trivial would it be to implement something that made that distinction simpler?

While the use of m4b files could help (similar to the podcast/URL stream implementation), many audiobooks have historically come as tracks of MP3 files, similar to an album of music. Perhaps in those cases it's just a matter of the user setting a flag somewhere to inform the jukebox app that the currently-playing content should be treated as an audiobook.

**Comment by votti** [comment]
@Seeking backwards and forward: Actually that is also a behaviour I wanted:
For the `seek forward` I found this already existed in the form of a `seek` function.
For `seek backwards` I implemented something to work both nicely for large and small audios:
https://github.com/votti/RPi-Jukebox-RFID/commit/c746000ec9d20172f93b96c867c4a3316f3c0592
Seeing that I am not the only one that likes this kind of behavior, I am opening a new issue.

**Comment by plimptm** [comment]
Very nice, thanks for sharing that commit.

Reading through your issue, I see that file length might have as much to do with the desired behavior than just media type.

How did you go about mapping your custom RPC call to a physical action? Did you use a custom control card, a physical switch, or something else?

**Comment by votti** [comment]
I mapped the custom RPC to a physical buttons (forward/backwards buttons on the box for the kids to smash).

**Comment by alesrosina** [comment]
Just wanted to check on this one - i see this is already implemented, but not sure exactly how to activate it? Do i need to manually edit `cards.yml`, so if i have sth like:

```yml
'871364165':
  alias: play_folder
  args:
  - Folder1
```

should i just add

```yml
'871364165':
  alias: play_folder
  args:
  - Folder1
  - resume: true
```

or how is this handeled? :)

Maybe it would be a cool addition to add this as checkbox in UI when adding a card.

**Comment by votti** [comment]
Currently setting this in the card config is exactly  what is required.

A tickbox in the UI would be definitely nice!
I used to hate UI work, so did not attempt it yet - but as AI loves UI and with your request there seems to be be at least some interest, I am happy to give it a try :) 

So far my lazy workaround was to simply set the default `resume = True` in my personal branch, and it is running well like this for me since 2024 :)

**Comment by alesrosina** [comment]
al right, so i'll just do this for all the cards. But there is no option to set this globally, right? Like in rfid.yaml?

**Comment by votti** [comment]
Currently there is no global option.

Also just to be clear: this does not work on the normal Jukebox Version, just on my branch:
https://github.com/votti/RPi-Jukebox-RFID/tree/future3/feature/resume

Which has an open request to merge into the normal version here:
https://github.com/MiczFlor/RPi-Jukebox-RFID/pull/1946

**Comment by alesrosina** [comment]
Right. So i would need to get it from your fork. Will try this at some point, but it seems exactly what i would need. to bad it's not yet merged :)
